src/store_client.cc

```diff
@@ -251,16 +252,12 @@ store_client::copy(StoreEntry * anEntry,
 #endif
 
     assert(!_callback.pending());
-#if ONLYCONTIGUOUSREQUESTS
-
-    assert(cmp_offset == copyRequest.offset);
-#endif
     /* range requests will skip into the body */
-    cmp_offset = copyRequest.offset;
     _callback = Callback (callback_fn, cbdataReference(data));
     copyInto.data = copyRequest.data;
     copyInto.length = copyRequest.length;
     copyInto.offset = copyRequest.offset;
```

Comment:
As we discussed, it is unclear what do copyInto fields mean here. 
For Http path:
**copyInto.data, copyInto.length** - the pointer to the client fixed-size array HttpStream::reqbuf and its size HTTP_REQBUF_SZ (4096).
**copyInto.offset** is the store offset, including range offset, i.e., it is what Http::Stream::getNextRangeOffset() returns.